src/Sly/NotificationPusher/Adapter/Gcm.php

```diff
@@ -67,15 +67,25 @@ public function push(PushInterface $push)
             $message = $this->getServiceMessageFromOrigin($tokensRange, $push->getMessage());
 
             try {
-                $this->response = $client->send($message);
-            } catch (ServiceRuntimeException $e) {
-                throw new PushException($e->getMessage());
-            }
 
-            if ((bool) $this->response->getSuccessCount()) {
-                foreach ($tokensRange as $token) {
-                    $pushedDevices->add($push->getDevices()->get($token));
+                $response = $client->send($message);
+                $responseResults = $response->getResults();
+
+                if ((bool) $response->getSuccessCount()) {
+                    foreach ($tokensRange as $token) {
+                        $device = $push->getDevices()->get($token);
+                        
+                        // map the overall response object
+                        // into a per device response
+                        $tokenResponse = @$responseResults[$token] ?: [];
```

Comment:
@theluk, it's not a good idea to use `@` to suppress warnings. Is there another way to check errors?